src/vec.rs

```diff
@@ -241,12 +236,27 @@ impl<T: Clone> EcoVec<T> {
         // Ensure unique ownership and grow the vector if necessary.
         self.reserve((self.len == self.capacity()) as usize);
 
+        // Safety: we just called `EcoVec::reserve()`
+        unsafe { self.push_unchecked(value) }
+    }
+
+    /// Add a value at the end of the vector, without reallocating.
+    ///
+    /// # Safety
+    /// Caller must ensure that `self.len < self.capacity()` and
+    /// `self.is_unique()` hold, such as by calling `EcoVec::reserve()`, or
+    /// `EcoVec::with_capacity()`.
+    #[inline]
+    pub unsafe fn push_unchecked(&mut self, value: T) {
```

Comment:
I wonder whether this method should really be public. `Vec` does not have this method. Can we make it private?

Reply:
I don't see any harm in it. It is marked `unsafe`, so any misuse is the caller's responsibility.

Reply:
Personally, I think a method should only be public if there's a strong case for it being useful. The fact that `Vec` does not expose such a method indicates to me that it is not typically needed. I think it's a good goal to keep a crate's public API surface as minimal as possible.